django/librarian/tasks.py

```diff
@@ -18,15 +19,15 @@
     fetch_from_url,
     get_process_engine_from_type,
 )
-from librarian.utils.vector_store_helpers import connect_to_vector_store
+from otto.models import User
 
 logger = get_logger(__name__)
 
 ten_minutes = 600
 
 
 @shared_task(soft_time_limit=ten_minutes)
-def process_document(document_id, language=None):
+def process_document(document_id, user_upn=None, language=None):
```

Comment:
Unlike the file translation Celery function, I had to pass a string for the user in here. 
You can't pass User objects directly since they aren't serializable.
And we can't infer the user from the Document since more than 1 user can have responsibility for a document.data_source.library.